s3/replication/replicator/tests/system/load_transfer_test.py

```diff
@@ -0,0 +1,295 @@
+        self._state = S3RequestState.COMPLETED
+
+
+def init_logger():
+    log_config_file = os.path.join(os.path.dirname(__file__),
+                                   'config', 'logger_config.yaml')
+
+    print("Using log config {}".format(log_config_file))
+    logger = setup_logger('client_tests', log_config_file)
+    if logger is None:
+        print("Failed to configure logging.\n")
+        sys.exit(-1)
+    return logger
+
+
+def create_replication_job(s3_config, object_info):
+    template_file = os.path.join(
+        os.path.dirname(__file__),
+        'data',
+        'replication_job_template.json')
+    with open(template_file, 'r') as file_content:
+        job_dict = json.load(file_content)
+    # Update the fields in template.
+    job_dict["replication-id"] = s3_config.source_bucket_name + \
+        object_info["object_name"]
+    epoch_t = datetime.datetime.utcnow()
+    job_dict["replication-event-create-time"] = epoch_t.strftime(
+        '%Y%m%dT%H%M%SZ')
+
+    job_dict["source"]["endpoint"] = s3_config.endpoint
+    job_dict["source"]["service_name"] = s3_config.s3_service_name
+    job_dict["source"]["region"] = s3_config.s3_region
+    job_dict["source"]["access_key"] = s3_config.access_key
+    job_dict["source"]["secret_key"] = s3_config.secret_key
+
+    job_dict["source"]["operation"]["attributes"]["Bucket-Name"] = \
+        s3_config.source_bucket_name
+    job_dict["source"]["operation"]["attributes"]["Object-Name"] = \
+        object_info["object_name"]
+    job_dict["source"]["operation"]["attributes"]["Content-Length"] = \
+        object_info["size"]
+    job_dict["source"]["operation"]["attributes"]["Content-MD5"] = \
+        object_info["md5"]
+
+    job_dict["target"]["endpoint"] = s3_config.endpoint
+    job_dict["target"]["service_name"] = s3_config.s3_service_name
+    job_dict["target"]["region"] = s3_config.s3_region
+    job_dict["target"]["access_key"] = s3_config.access_key
+    job_dict["target"]["secret_key"] = s3_config.secret_key
+
+    job_dict["target"]["Bucket-Name"] = s3_config.target_bucket_name
+
+    return job_dict
+
+
+async def async_put_object(session, bucket_name, object_name, object_size,
+                           transfer_chunk_size):
+
+    object_reader = ObjectDataGenerator(session.logger,
+                                        object_name, object_size)
+    object_writer = S3AsyncPutObject(session, bucket_name,
+                                     object_name, object_size)
+
+    # Start transfer
+    await object_writer.send(object_reader, transfer_chunk_size)
+    return {"object_name": object_name, "size": object_size,
+            "md5": object_reader.get_md5(),
+            "etag": object_writer.get_response_header("ETag")
+            }
+
+
+async def setup_source(session, bucket_name, transfer_chunk_size):
+    # Create objects at source and returns map with object info
+
+    # {"object-name1": {size: 4096, md5: abcd},
```

Comment:
in this example, may be we want to use keys inside double quotes :

i.e. {"object-name1" : { "size": 4096, "md5": "abcd"},
"object-name2" : { "size": 4192, "md5": "pqr"}}

Reply:
Done